src/forms/EditListingAudioForm/EditListingAudioForm.js

```diff
@@ -0,0 +1,232 @@
+          // before creating a new listing. If that order is changed, these should be changed too.
+          // Create and show listing errors are shown above submit button
+          const publishListingFailed = publishListingError ? (
+            <p className={css.error}>
+              <FormattedMessage id="EditListingAudioForm.publishListingFailed" />
+            </p>
+          ) : null
+          const showListingFailed = showListingsError ? (
+            <p className={css.error}>
+              <FormattedMessage id="EditListingAudioForm.showListingFailed" />
+            </p>
+          ) : null
+
+          const submittedOnce = this.submittedAudio.length > 0
+          // audio can contain added audios (with temp ids) and submitted audios with uniq ids.
+          const arrayOfAudioIds = (audio) =>
+            audio.map((i) => (typeof i.id === 'string' ? i.audioId : i.id))
+          const audioIdsFromProps = arrayOfAudioIds(audio)
+          const audioIdsFromPreviousSubmit = arrayOfAudioIds(this.submittedAudio)
+          const audioArrayHasSameImages = isEqual(audioIdsFromProps, audioIdsFromPreviousSubmit)
+          const pristineSinceLastSubmit = submittedOnce && audioArrayHasSameImages
+
+          const submitReady = (updated && pristineSinceLastSubmit) || ready
+          const submitInProgress = updateInProgress
+          const submitDisabled =
+            invalid || disabled || submitInProgress || audioUploadRequested || ready
+
+          const classes = classNames(css.root, className)
+
+          const listAudio = this.submittedAudio.length > 0 ?
+            (<ul>
+              { this.submittedAudio.map((fileName, index) =>
+                <li key={fileName + index}>
+                  {fileName}
+                  { this.state.currentFileName === fileName ? <div style={{width: 200, height: 15, border: '1px solid red'}}>
+                    <div style={{width: this.state.currentFileProgress * 2, height: 15, background:'green'}}></div>
+                  </div>:<div style={{width: 200, height: 15, background:'green'}}></div>}
+                </li>)
+              }
+            </ul>) : <></>
+
+          return (
+            <Form
+              className={classes}
+              onSubmit={(e) => {
+                bypassHandleSubmit({ audio: this.submittedAudio})
+                e.preventDefault()
+              }}
+            >
+              {updateListingError ? (
+                <p className={css.error}>
+                  <FormattedMessage id="EditListingAudioForm.updateFailed" />
+                </p>
+              ) : null}
+
+              { listAudio }
+
+              <ReactS3Uploader
+                signingUrl="/s3/sign"
+                signingUrlMethod="GET"
+                accept={ACCEPT_AUDIO}
+                // preprocess={(thing)=> console.log({thing})}
+                onSignedUrl={(res) => {
+                  this.handleUploadProgress({ type: 'fileName', fileName: res.filename })
```

Comment:
Remind me over whiskey and I'll show you a trick for this kind of thing using closures. 🥃